src/main/java/io/lettuce/core/ReadFromImpl.java

```diff
@@ -129,6 +136,60 @@ public ReadFromAnyReplica() {
 
     }
 
+    /**
+     * Read from any node in the subnets.
+     *
+     * @since x.x.x
+     */
+    static final class ReadFromSubnet extends ReadFrom {
+
+        private final List<IpSubnetFilterRule> rules = new ArrayList<>();
+
+        /**
+         * @param cidrNotations CIDR-block notation strings, e.g., "192.168.0.0/16".
+         */
+        ReadFromSubnet(String... cidrNotations) {
+            LettuceAssert.notEmpty(cidrNotations, "cidrNotations must not be empty");
+
+            for (String cidrNotation : cidrNotations) {
+                // parts[0]: ipAddress (e.g., "192.168.0.0")
+                // parts[1]: cidrPrefix (e.g., "16")
+                String[] parts = cidrNotation.split("/");
+                LettuceAssert.isTrue(parts.length == 2, "cidrNotation must have exact one '/'");
+
+                rules.add(new IpSubnetFilterRule(parts[0], Integer.parseInt(parts[1]), IpFilterRuleType.ACCEPT));
+            }
+        }
+
+        @Override
+        public List<RedisNodeDescription> select(Nodes nodes) {
+            List<RedisNodeDescription> result = new ArrayList<>(nodes.getNodes().size());
+            for (RedisNodeDescription node : nodes) {
+                if (isInSubnet(node.getUri())) {
```

Comment:
[Question] As far as I tried with debugger, `node.getUri()`  always returns IP address rather than hostname like not `redis://foobar.com` but `redis://192.168.0.1`.
Is there any case of returning hostname here?
My concern is, if there is any case that `node.getUri()` returns initial endpoint, this code may not work as we intend because the initial endpoint may be associated with multiple IP addresses.

For AWS specific, they issue configuration endpoint (e.g., `foobar.cache.amazonaws.com`)  and we use it for initial endpoint. The endpoint is associated with (almost all) cluster nodes' ip address. 
```
$ dig foobar.cache.amazonaws.com +short
192.168.0.1
192.168.0.2
192.168.0.3
```
So there is any case that `node.getUri()` returns initial endpoint, its IP address is any one of cluster nodes', thus this class does not work as we intend.

Reply:
The host part of `RedisURI` depends on from where the URI was supplied. In a Redis Cluster or Redis Sentinel arrangement, the topology source is always Redis itself which reports IP addresses only.

When using Master/Replica with provided URL's, then you get hold of the hostname.

`ReadFrom` gets always called with the target node. Does that help?

Reply:
Thank you for the information and review. I will investigate the behaviour in detail!